src/main/java/com/neroreflex/plugins/HackerNews.java

```diff
@@ -31,64 +31,62 @@
  */
 public final class HackerNews extends Trancio {
 
-    private int interval = 30000; //tempo tra le chiamate all'api in millisecondi
+    private int interval = (int)2.16e+7; // (= 6h) tempo tra le chiamate all'api in millisecondi
 
     private final String apiBaseURL = "https://hacker-news.firebaseio.com/v0/";
     private final String apiTopStories = apiBaseURL + "topstories.json";
     private final String apiItem = apiBaseURL + "item/";
-
-    private ArrayList<Integer> topStories = new ArrayList<Integer>();
-
-    protected String onHelp() {
-      return "No commands are available. The plugin will automatically notify new posts on Hacker News.";
-    }
+    private final String defaultURL = "https://news.ycombinator.com/item?id=";
 
     private JsonStructure jsonApiCall(String endpoint) throws IOException{
-        JsonReader rdr = Json.createReader(new URL(endpoint).openStream());
-        JsonStructure result = rdr.read();
-        rdr.close(); //questa chiamata chiude anche l'InputStream sottostante  (http://docs.oracle.com/javaee/7/api/javax/json/JsonReader.html#close--)
-        return result;
+      JsonReader rdr = Json.createReader(new URL(endpoint).openStream());
+      JsonStructure result = rdr.read();
+      rdr.close(); //questa chiamata chiude anche l'InputStream sottostante  (http://docs.oracle.com/javaee/7/api/javax/json/JsonReader.html#close--)
+      return result;
     }
 
-    //Notifica nella chat la notizia con l'id specificato.
-    private void notifyNews(int id) throws IOException{
-        JsonObject obj = (JsonObject)jsonApiCall(apiItem + id + ".json"); //apre la connessione con l'api
-        if(obj.getString("type").equals("story")){ //controlla che sia una story (un thread) e non un comento, poll, ecc.
-            String[] channels = getChannels();
-            for(String chan: channels)
-                sendMessage(new Message(chan, "From Hacker News: " + obj.getString("title") + " " + obj.getString("url")));
+    private class NewsEntry implements Comparable<NewsEntry>{
+        private int score;
+        private String title;
+        private String url;
+        public NewsEntry(int id) throws IOException{
+            JsonObject obj = (JsonObject)jsonApiCall(apiItem + id + ".json"); //apre la connessione con l'api
+            score = obj.getInt("score");
+            title = obj.getString("title");
+            try{
+                url = obj.getString("url");
+            }catch(NullPointerException e){
+                url = defaultURL + id;
+            }
+        }
+        public int compareTo(NewsEntry other){
+            return other.score - score;
+        }
+        public String toString(){
+            return "From Hacker News: " + title + " " + url + " (score = " + score + ")";
```

Comment:
Forse si può togliere lo score, in alternativa si può lasciare solamente il numero tra parentesi, segnalare che si tratta dello score non è molto importante e aggiunge caratteri (su IRC lo spazio è prezioso)